Sources/Sentry/SentryEnvelope.m

```diff
@@ -198,6 +206,35 @@ - (_Nullable instancetype)initWithAttachment:(SentryAttachment *)attachment
     return [self initWithHeader:itemHeader data:data];
 }
 
+- (instancetype)initWithReplayEvent:(SentryReplayEvent *)replayEvent
+                    replayRecording:(SentryReplayRecording *)replayRecording
+                              video:(NSURL *)videoURL
+{
+    NSData *replayEventData = [SentrySerialization dataWithJSONObject:[replayEvent serialize]];
+    NSMutableData *recording = [NSMutableData data];
+    [recording appendData:[SentrySerialization
+                              dataWithJSONObject:[replayRecording headerForReplayRecording]]];
+    [recording appendData:[SentrySerialization dataWithJSONObject:[replayRecording serialize]]];
+
+    NSURL *envelopeContentUrl =
+        [[videoURL URLByDeletingPathExtension] URLByAppendingPathExtension:@"dat"];
+
+    [SentryMsgPackSerializer serializeDictionaryToMessagePack:@{
+        @"replay_event" : replayEventData,
+        @"replay_recording" : recording,
+        @"replay_video" : videoURL
+    }
+                                                     intoFile:envelopeContentUrl];
+
+    // TODO: Create and envelope item that accepts and URL as content so we dont need to load the
```

Comment:
I think it's acceptable to release a version without this improvement. I would definitely not add this improvement to this PR.

Reply:
Yes, Im not adding to this PR.
Maybe on a follow up PR.